esi_leap/objects/lease.py

```diff
@@ -211,3 +309,11 @@ def resource_object(self):
     def verify_child_availability(self, start_time, end_time):
         return self.dbapi.lease_verify_child_availability(
             self, start_time, end_time)
+
+    def _get_resource_info(self, resource):
+        resource._node_name = resource.get_name()
+        resource._provision_state \
+            = resource.get_node_provision_state()
+        resource._power_state = resource.get_node_power_state()
+        resource._node_properties = resource.get_config()
+        return resource
```

Comment:
How are these values pulled out by the notification? Setting and using `_` variables is probably not idea, when methods like `resource.get_name()` retrieve the value (and exists for all three node types).

Reply:
The ironic_node and lease objects are passed to populate_schema() https://github.com/DanNiESh/esi-leap/blob/events/esi_leap/objects/notification.py#L132. Then it uses getattr() method to get the node and lease objects' attribute ( _power_state, _provision_state, _uuid, etc.) based on the schema.

Reply:
In that case, I might suggest creating something like a new 'resource_event' object that's specifically meant to be passed in, and have this function just instantiate that object.

Reply:
Will do